odoo/addons/test_access_rights/tests/test_feedback.py

```diff
@@ -398,6 +398,40 @@ def test_warn_company_access(self):
         ):
             p.with_user(self.user).val
 
+    def test_warn_company_access_multi_record(self):
+        """ because of prefetching, read() goes through a different codepath
+        to apply rules
+        """
+
+        company1_id = self.env['res.company'].create({'name': 'Brosse Inc.'})
+        company2_id = self.env['res.company'].create({'name': 'Brosse Inc. 2'})
+        record_1 = self.env["test_access_right.some_obj"].create({"val": 5, "company_id": company1_id.id})
+        record_2 = self.env["test_access_right.some_obj"].create({"val": 6, "company_id": company2_id.id})
+        self.user.sudo().company_ids = [Command.link(company1_id.id)]
+        self.user.sudo().company_ids = [Command.link(company2_id.id)]
+        self._make_rule('rule 0', "[('company_id', 'in', [False])]", attr='read')
+        with self.debug_mode(), self.assertRaises(AccessError) as ctx:
+            records = record_1 + record_2
+            self.env.invalidate_all()
+            _ = records.with_user(self.user).read(["val"])
+        self.maxDiff = None
+        self.assertEqual(
+            ctx.exception.args[0],
+            """Uh-oh! Looks like you have stumbled upon some top-secret records.
+
+Sorry, %s (id=%s) doesn't have 'read' access to:
+- %s, %s (%s: %s, company=%s)
+ - %s, %s (%s: %s, company=%s)
```

Comment:
Can you remove this useless extra begin space here by changing `failing_records = ' \n'...` => `failing_records = '\n'...`.